### PR TITLE
git: add Linux dependency and make some parts macOS-only

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
This is not a complete upstreaming of the Linux changes for git - just some of the simple stuff that shouldn't require a CI run for now.

File: Formula/git.rb [hunk 1]
```diff
@@ -22,6 +22,14 @@ class Git < Formula
   depends_on "gettext"
   depends_on "pcre2"
 
+  on_linux do
+    depends_on "curl"
+    depends_on "expat"
+    depends_on "linux-headers"
+    depends_on "openssl@1.1"
+    depends_on "zlib"
+  end
```
Comment (carlocab) [inline Formula/git.rb:31]:
```suggestion
  uses_from_macos "curl"
  uses_from_macos "expat"
  uses_from_macos "zlib"

  on_linux do
    depends_on "linux-headers"
    depends_on "openssl@1.1"
  end
```

Comment (danielnachun) [reply Formula/git.rb:31]: Actually, I think `uses_from_macos "openssl@1.1"` is also allowed, so I can change that as well.

File: Formula/git.rb [hunk 2]
```diff
@@ -85,12 +96,14 @@ def install
     git_core = libexec/"git-core"
 
     # Install the macOS keychain credential helper
-    cd "contrib/credential/osxkeychain" do
-      system "make", "CC=#{ENV.cc}",
-                     "CFLAGS=#{ENV.cflags}",
-                     "LDFLAGS=#{ENV.ldflags}"
-      git_core.install "git-credential-osxkeychain"
-      system "make", "clean"
+    on_macs do
```
Comment (carlocab) [inline Formula/git.rb:99]:
```suggestion
    on_macos do
```
Let's run this through CI in case there are other subtle errors I missed 🙂

**Review by carlocab — APPROVED** [review]

**Pull request merged**